EIPS/eip-2583.md

```diff
@@ -0,0 +1,295 @@
+| CREATE2  | No | Create destination not explicitly settable, and assumed to be inexistent already.| 
+
+
+### Notes on Call-derivatives
+
+
+A `CALL` triggers a lookup of the `CALL` destination address. The base cost for `CALL` is at `700` gas. A few other characteristics determine the actual gas cost of a call:
+
+1. If the `CALL` (or `CALLCODE`) transfers value, an additional `9K` is added as cost. 
+    1.1 If the `CALL` destination did not previously exist, an additional `25K` gas is added to the cost.
+
+This EIP adds a second rule in the following way:
+
+2. If the call does _not_ transfer value and the callee does _not_ exist, then `penalty` gas is added to the cost.
+
+In the table below, 
+- `value` means non-zero value transfer, 
+- `!value` means zero value transfer, 
+- `dest` means destination already exists, or is a `precompile`
+- `!dest` means destination does not exist and is not a `precompile`
+
+| Op | value,dest| value, !dest |!value, dest| !value, !dest|  
+| -- | ---------  | -- | --| -- |
+|CALL |  no change | no change| no change| `penalty`|
+|CALLCODE |  no change | no change| no change| `penalty`|
+|DELEGATECALL | N/A | N/A| no change| `penalty` |
+|STATICCALL | N/A | N/A| no change| `penalty` |
+
+Whether the rules of this EIP is to be applied for regular ether-sends in `transactions` is TBD. See the 'Backwards Compatibility'-section for some more discussion on that topic. 
+
+### Note on `SELFDESTRUCT`
+
+The `SELFDESTRUCT` opcode also triggers an account trie lookup of the `beneficiary`. However, due to the following reasons, it has been omitted from having a `penalty` since it already costs `5K` gas. 
+
+### Clarifications:
+
+- The `base` costs of any opcodes are not modified by the EIP.
+- The opcode `SELFBALANCE` is not modified by this EIP, regardless of whether the `self` address exists or not. 
+
+### Determining the `penalty`
+
+A transaction with `10M` gas can today cause ~`14K` trie lookups.
+
+- A `penalty` of `1000`would lower the number to ~`5800` lookups, `41%` of the original.
+- A `penalty` of `2000`would lower the number to ~`3700` lookups, `26%` of the original.
+- A `penalty` of `3000`would lower the number to ~`2700` lookups, `20%` of the original. 
+- A `penalty` of `4000`would lower the number to ~`2100` lookups, `15%` of the original. 
+
+There exists a roofing function for the `penalty`. Since the `penalty` is deducted from `gas`, that means that a malicious contract can always invoke a malicious relay to perform the trie lookup. Let's refer to this as the 'shielded relay' attack. 
+
+In such a scenario, the `malicious` would spend `~750` gas each call to `relay`, and would need to provide the `relay` with at least `700` gas to do a trie access.
+
+Thus, the effective `cost` would be on the order of `1500`. It can thus be argued that `penalty` above `~800` would not achieve better protection against state-inexistence attacks. 
```

Comment:
This should be in the Rationale section.  The rationale is where you discuss *why* you chose the constants you chose.  The specification section should merely assert that the constant is X and it is applied here, here, and here.